src/components/CompareResults/ResultsMain.tsx

```diff
@@ -52,6 +52,7 @@ function ResultsMain() {
             <CircularProgress />
           </Box>
         }
+        key={Date.now()}
```

Comment:
I'm not fully sure this is necessary, we might revisit later when we'll manage to cut the rendering time down.

Reply:
The rationale is that the Suspense component will be remounted, and so doesn't keep a state. It seemed like it improved things with my limited testing...

Reply:
So wait, without this key, the spinner was _not_ appearing when doing updates (changing revisions and frameworks) on the Results page?

Reply:
I ask because after rebasing and resolving your comments on my PR, the spinner appears on updates and framework updates.

Reply:
@Carla-Moz the fact I'm using "key" is backed by my experiment that you can find in https://codepen.io/julienw/pen/MWMooEB

This experiment is built to replicate a bit how perfcompare behaves currently:
* the loader itself awaits for a timeout of 500ms, this models the revision info fetches
* it returns a promise that resolves 200ms later. This models the compare results fetch.

These timings are important, because the behavior change if you change them (more on that below).

You can see that Suspense has "key" in it. If you press "refresh", we can see "Loading results".
However if you remove the "key", save and wait for the reload, then "refresh" doesn't show "Loading results" anymore.


This behavior is different if you tweak the timings. For example if you totally remove the initial timeout of 500ms, then the fallback always appears, even without the `key`.
Similarly, if you increase the returned promise timeout to more than 500ms, then you see it as well.
My guess (that should be verified by looking at the code) is that Suspense waits for at least the time for the loader to return before showing the fallback. But I'm not sure which library is responsible for that, is that react-router-dom or React itself, or the interaction of them both? I've no idea.

Reply:
Thanks for explaining. That makes a lot more sense.